### PR TITLE
Fix for #461

### PR DESCRIPTION
**Pull request by certik**

### CODE REVIEW

File: symengine/integer.h [hunk 1]
```diff
@@ -152,12 +152,13 @@ struct RCPIntegerKeyLess
         return false;
     }
 };
-//! \return RCP<const Integer> from `int`
-inline RCP<const Integer> integer(int i)
+//! \return RCP<const Integer> from integral values
+template<typename T> inline
+typename std::enable_if<std::is_integral<T>::value, RCP<const Integer>>::type
```
Comment (certik) [inline symengine/integer.h]: Do we still need this `enable_if` in here? It seems it would work without it as well.

Comment (mafiya69) [inline symengine/integer.h]: Also I think, we don't need the another overload for `integer` function. So I am removing `std::enable_if` and another overload of `integer`.

Comment (certik) [inline symengine/integer.h]: Agreed

**Comment by mafiya69** [comment]
@certik Anything more to do?

**Comment by certik** [comment]
No, this looks great! Thanks for improving the code.

**Pull request merged**